pkg/apis/discovery/validation/validation.go

```diff
@@ -110,6 +120,17 @@ func validateEndpoints(endpoints []discovery.Endpoint, addrType discovery.Addres
 			}
 		}
 
+		if endpoint.NodeName != nil {
```

Comment:
I never really understood why we check the gate in validation. I see three possible scenarios here:
1. the gate was on and we didn't drop it in strategy -- no need to check `allowNodeName` here since strategy allowed the field to be non-nil.
2. the gate is off and the field was dropped -- this check would be skipped since it's nil
3. the gate is off and on roll back a resource has the field set -- this is identical to 1. in terms of validation logic.

Reply:
Yeah I'm not sure if this really has any value since the strategy should cover everything. I have seen this pattern in a number of validation files though, so I'd lean towards leaving it in just in case we're missing anything.

Reply:
We should not be checking gates except rare cases

Reply:
Removed, thanks!